sphinx/domains/c.py

```diff
@@ -197,21 +197,23 @@ def get_index_text(self, name: str) -> str:
             return ''
 
     def add_target_and_index(self, name: str, sig: str, signode: desc_signature) -> None:
-        # for C API items we add a prefix since names are usually not qualified
-        # by a module name and so easily clash with e.g. section titles
-        targetname = 'c.' + name
-        if targetname not in self.state.document.ids:
-            signode['names'].append(targetname)
```

Comment:
I think mention in CHANGES is not needed for C domain. Because it might be not official and hidden feature.
```
.. c:function:: func

ref with `c.func`_
```